### PR TITLE
Introduce ResetTest, use in CL for 2x test speedup

### PR DESCRIPTION
**Pull request by ValarDragon**
This introduces a method called `ResetTest` which avoids having to repeatedly make NewOsmosisApp's and run InitGenesis every call.

Takes tests on my machine for CL from 11.4 seconds to 4.1s. Attached pprof below:

EDIT: now 3.8s, missed some things that should've been reset's

(This is before roman's change for reducing duplication got in)

Before:
image

After:
image

### CODE REVIEW

**Comment by ValarDragon** [comment]
Update, did a driveby speedup to the invariant test, now at 1.9s

File: app/apptesting/test_suite.go [hunk 1]
```diff
@@ -43,6 +43,8 @@ import (
 
 type KeeperTestHelper struct {
 	suite.Suite
+	hasUsedAbci bool
+	withCaching bool // defaults to false
```
Comment (p0mvn) [inline app/apptesting/test_suite.go:47]: Could you add comments around when we expect these to be toggled please?

**Review by p0mvn — APPROVED** [review]
Awesome! Great work

File: app/apptesting/test_suite.go (hunk 1 above)
Comment (ValarDragon) [reply app/apptesting/test_suite.go]: done!

**Pull request merged**